src/code/z_player_lib.c

```diff
@@ -858,7 +858,7 @@ s32 func_8008FCC8(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* p
     if (limbIndex == PLAYER_LIMB_ROOT) {
         D_80160014 = this->leftHandType;
         D_80160018 = this->rightHandType;
-        D_80160000 = &this->swordInfo[2].base;
+        sCurBodyPartPos = &this->bodyPartsPos[-1];
```

Comment:
Hmm, does this lead to oob reads anywhere?

Reply:
only way this will ever stay null is if player has no dlists at all for any limb in his skeleton (wont happen)
agree its sketch though

Reply:
Might be worth noting that, then, especially given that #1142 has an example of an actual oob read in the same file.

Reply:
yeah it's incredibly sketch

this will work correctly as long as you don't touch link's skeleton basically, since the bodypart indices rely on both limb order and which limbs have dlists attached or not

In `func_8008FCC8` (the common overridelimbdraw):
- `sCurBodyPartPos = &this->bodyPartsPos[-1];` happens when the limb is root (first processed limb)
- `sCurBodyPartPos++` happens when the limb isn't root and the limb's dlist is non-NULL

In `func_80090D20` (postlimbdraw):
- `Matrix_MultVec3f(&sZeroVec, sCurBodyPartPos);` if the limb's dlist is non-NULL

so if the root limb had a dlist then yeah `this->bodyPartsPos[-1]` would be written to

It's easier to follow with the skelanime callbacks named from #1098, I don't think it needs an explanation, does it?

Reply:
Ah, that's what this is about, I see, thanks. Well, it probably is worth pointing out that what looks like it might cause an oob read, actually it will not provided that ROOT doesn't have a dlist. But I do feel like at this point indexing with -1 kind of needs a comment either way.